src/app/shared/dso-selector/modal-wrappers/dso-selector-modal-wrapper.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, Injectable, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { DSpaceObject } from '../../../core/shared/dspace-object.model';
+import { RemoteData } from '../../../core/data/remote-data';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { map } from 'rxjs/operators';
+import { DSpaceObjectType } from '../../../core/shared/dspace-object-type.model';
+
+export enum SelectorActionType {
+  CREATE = 'create',
+  EDIT = 'edit'
+}
+
+@Injectable()
+export abstract class DSOSelectorModalWrapperComponent implements OnInit {
```

Comment:
This class should have a description in TypeDocs, as it seems important to describe its purpose (since all the various components in this PR depend on it)

Reply:
Added missing TypeDoc